roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,76 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class DockerImageAvailability(NotContainerizedMixin, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    # attempt to pull required docker images and fail if
+    # any images are missing, or error occurs during pull
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+        openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+
+        changed = False
+        failed_pulls = set()
+
+        pulled_images = self.pull_images(self.add_image_tags(self.rpm_docker_images(), openshift_release), tmp, task_vars)
+        if "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            if len(failed_pulls) < len(self.rpm_docker_images()) + len(self.rpm_qualified_docker_images()):
+                changed = True
+            return {"changed": changed, "failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.rpm_qualified_docker_images()))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0)
+        pulled_qualified_images = self.pull_images(self.add_image_tags(self.rpm_qualified_docker_images(), "v" + openshift_image_tag), tmp, task_vars)
```

Comment:
@detiber, can you explain what you mean by "allow us to be a bit fuzzier on finding a matching container image"?

@juanvallejo, thanks for digging in to skopeo.  We need to find the right balance between idempotence and usefullness.  Here's a question, is there already a skopeo container?  I _really_ don't want to require our checks to need any sort of `subscription-manager` work to register a system to pull RPMs.  Until skopeo is somehow built in it seems OK in my mind to pull a container in order to run it.

 @rhcarvalho's comments are still valid.  We'd have to be completely convinced we aren't going to have skopeo report that the correct images are available only to have a docker pull fail because a misconfigured registry is shadowing registry.access.redhat.com.

@juanvallejo, To test this further I'd be curious to know what happens if the same repository exists in two registries yet a specific tag only exists in a "shadowed" registry configured with docker.  To test you could set up a registry using the steps at https://docs.docker.com/registry/deploying/ that exposed an image called /openshift3/ose-pod:latest.  Then configure that registry to be used by a docker pull with `ADD_REGISTRY` in `/etc/sysconfig/docker`.  Put it _before_ the entry for registry.access.redhat.com.  Then make sure `docker pull openshift3/ose-pod:v3.4` still works.  It should first check the misconfigured registry and see that the image is missing and precede to check registry.access.redhat.com.